packages/@glimmer/compiler/lib/passes/1-normalization/keywords/expr.ts

```diff
@@ -36,15 +36,6 @@ export const EXPR_KEYWORDS = keywords('Expr')
   })
   .kw('component', {
     assert(node: ExprKeywordNode): Result<{ definition: ASTv2.ExpressionNode; args: ASTv2.Args }> {
-      if (node.type !== 'Call') {
-        return Err(
-          generateSyntaxError(
-            'The (component) keyword must be called with arguments in order to curry a component definition. It cannot be used directly as a value.',
-            node.loc
-          )
-        );
-      }
```

Comment:
This does still error, with a better error message now. The difference is that keywords no longer run on path references at all, so it will always be `node.type === 'Call'`. The error message is thrown when we iterate through PathReference